Objects/templateobject.c

```diff
@@ -344,47 +297,17 @@ template_concat_templates(templateobject *self, templateobject *other)
     return newtemplate;
 }
 
-static PyObject *
-template_concat_template_str(templateobject *self, PyObject *other)
-{
-    PyObject *newstrings = template_strings_append_str(self->strings, other);
-    if (newstrings == NULL) {
-        return NULL;
-    }
-
-    PyObject *newtemplate = _PyTemplate_Build(newstrings, self->interpolations);
-    Py_DECREF(newstrings);
-    return newtemplate;
-}
-
-static PyObject *
-template_concat_str_template(templateobject *self, PyObject *other)
-{
-    PyObject *newstrings = template_strings_prepend_str(self->strings, other);
-    if (newstrings == NULL) {
-        return NULL;
-    }
-
-    PyObject *newtemplate = _PyTemplate_Build(newstrings, self->interpolations);
-    Py_DECREF(newstrings);
-    return newtemplate;
-}
-
 PyObject *
 _PyTemplate_Concat(PyObject *self, PyObject *other)
 {
     if (_PyTemplate_CheckExact(self) && _PyTemplate_CheckExact(other)) {
         return template_concat_templates((templateobject *) self, (templateobject *) other);
     }
-    else if ((_PyTemplate_CheckExact(self)) && PyUnicode_Check(other)) {
-        return template_concat_template_str((templateobject *) self, other);
-    }
-    else if (PyUnicode_Check(self) && (_PyTemplate_CheckExact(other))) {
-        return template_concat_str_template((templateobject *) other, self);
-    }
-    else {
-        Py_RETURN_NOTIMPLEMENTED;
-    }
+
+    PyErr_Format(PyExc_TypeError,
+        "can only concatenate Template (not \"%T\") to Template",
+        other);
```

Comment:
Let's include the full path to `Template` since just `Template` is ambiguous.

```suggestion
    PyErr_Format(PyExc_TypeError,
        "can only concatenate string.templatelib.Template (not \"%T\") to string.templatelib.Template",
        other);
```